megatron/optimizer/distrib_optimizer.py

```diff
@@ -542,7 +542,11 @@ def reduce_model_grads(self, args, timers):
         timers('backward-embedding-all-reduce').stop()
 
         # All-reduce key-value grads if needed.
-        if args.attention_head_type == "multiquery":
+        if (
+            args.attention_head_type == "multiquery"
+            and mpu.get_tensor_model_parallel_world_size() > 1
```

Comment:
Maybe for consistency add this if condition in the `allreduce` method?

Reply:
I know the others do it but this is better because the timers have a big overhead.

Reply:
Actually you only need to sync when it's sequence parallel no?

Reply:
It also calls `torch.cuda.synchronize`, that's a lot worse